packages/@aws-cdk/aws-ec2/lib/user-data.ts

```diff
@@ -484,6 +486,16 @@ export class MultipartUserData extends UserData {
     this.addPart(MultipartBody.fromUserData(userData, contentType));
   }
 
+  /**
+   * Adds a multipart part based on a UserData object. This UserData is always added
+   * as a shell script. The UserData added by this method will also be the target
+   * of calls to the `add*Command` methods on this MultipartUserData object.
+   */
+  public addUserDataPartForCommands(userData: UserData) {
```

Comment:
I think I would prefer the first invocation of `adduserDataPart()` to trigger this behavior, optionally gated by a `makeDefault?: boolean` parameter (which defaults to `true` if no default has been set yet and defaults to `false` if a default has already been set).

This additional method also feels awkward.